fairscale/experimental/wgit/signal_sparsity.py

```diff
@@ -250,32 +261,29 @@ def dense_sst_to_dst(self, dense: Tensor, sst: Tensor) -> Tensor:
             (Tensor):
                 Same shaped tensor, still dense format but has zeros. Non-zeros are top-k delta values.
         """
-        pass
+        delta = dense - self.sst_dst_to_dense(sst)  # sst_dst_to_dense(sst) returns the inverse transform here
+        top_k_total_size = _top_k_total_size(dense, self._dst_top_k_dim)
+        k = _get_k_for_topk(self._dst_top_k_percent, self._dst_top_k_element, top_k_total_size)
+        return _scatter_topk_to_sparse_tensor(delta.abs(), delta, k, dim=self._dst_top_k_dim)
 
-    def sst_dst_to_dense(self, sst: Tensor, dst: Tensor = None) -> Tensor:
-        """From SST and dst back to a dense
-
-        result = ifft(sst)
-        if dst is not None:
-            result += dst
-        return result
+    def sst_dst_to_dense(self, sst: Tensor, dst: Optional[Tensor] = None) -> Tensor:
+        """From SST and DST returns a dense reconstruction. When argument dst=None, simply returns
+        the inverse transform of the SST tensor.
 
         Args:
             sst (Tensor):
                 Singal sparse tensor. Required argument.
-            dst (Tensor, optinoal):
+            dst (Tensor, optional):
                 Delta sparse tensor, optional.
 
         Returns:
             (Tensor):
                 A dense tensor in real number domain from the SST.
         """
-        pass
-
-    def sst_or_dst_to_mask(self) -> None:
-        # we shouldn't need this function since going from SST/DST to mask should be a
-        # trivial call in pytorch. Maybe I am missing something.
-        pass
+        dense_recons = self._inverse_transform(sst).real
```

Comment:
do FFT and DCT both have .real after inverse? 

Also, should inverse function always return the real part since the complex part should be zero, right?

If so, just absorb the `.real` part into the inverse transform function and caller will be simpler.

Reply:
with the DCT, we would either have to implement it with native pytorch operations, or use scipy. I don't think it would have a `.real` attribute by default, or we wanted to implement DCT as a class. Either way, seems like absorbing `real` would be better. And DCT is always real if I am not wrong.

Reply:
I think we can simply do `dense_recons = torch.real(self._inverse_transform(sst))` which will tackle both cases